### PR TITLE
1503 task3 secure web api with authentication and authorization

### PR DESCRIPTION
**Pull request by mjunaid-cnc**

### CODE REVIEW

File: Todo.Application/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Todo.Application
+{
+    public class Class1
```
Comment (Gulzaib83) [inline Todo.Application/Class1.cs]: Please check, if this class is not being used then remove it. Same goes for all such classes that are created by default by IDE and are not being used

File: Todo.Infrastructure/Services/TodoService.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        public async Task<TodoList?> GetItemById(Guid id)
+        {
+            try
+            {
+                var role = _userResolverService.GetUserRole();
+                var userId = _userResolverService.GetUserId();
+                TodoList? item;
+                if (string.Equals(role, UserRoleType.User.ToString()))
+                {
+                    item = await _context.TodoLists.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId);
```
Comment (Gulzaib83) [inline Todo.Infrastructure/Services/TodoService.cs:56]: Please design and use some Middleware to intercept the Call to particular EndPoint. This Middle ware shall be responsible to allow / restrict user from accessing certain endpoints on the basis of resource ownership

File: Todo.Application/Class1.cs (hunk 1 above)
Comment (mjunaid-cnc) [reply Todo.Application/Class1.cs]: Done

File: Todo.Infrastructure/Services/TodoService.cs (hunk 2 above)
Comment (mjunaid-cnc) [reply Todo.Infrastructure/Services/TodoService.cs:56]: Done. Added middleware for checking user access

Comment (mjunaid-cnc) [reply Todo.Infrastructure/Services/TodoService.cs:56]: I have removed the middleware and added an Action Filter for checking user access

**Review by Gulzaib83 — APPROVED** [review]
Code is fine, fulfilling requirements.

**Pull request merged**